damnit/ctxsupport/ctxrunner.py

```diff
@@ -14,24 +14,23 @@
 import pickle
 import sys
 import time
-from datetime import timezone
 import traceback
+from datetime import timezone
 from enum import Enum
+from graphlib import CycleError, TopologicalSorter
 from pathlib import Path
 from unittest.mock import MagicMock
-from graphlib import CycleError, TopologicalSorter
-
-from matplotlib.axes import Axes
-from matplotlib.figure import Figure
 
 import extra_data
 import h5py
 import numpy as np
-import xarray as xr
 import requests
+import xarray as xr
 import yaml
-
 from damnit_ctx import RunData, Variable
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+from plotly.graph_objects import Figure as PlotlyFigure
```

Comment:
We need to document the new dependency: https://damnit.readthedocs.io/en/latest/backend/#using-custom-environments

Reply:
Alternatively, we could do the check without making plotly a dependency, using a version of the `isinstance_no_import` function we've got in a couple of places already:

https://github.com/European-XFEL/EXtra/blob/7d0f66f781321f84b73790f27ccba11854a7608a/src/extra/components/utils.py#L30-L36

(I got this technique from IPython)